lib/utils/is-target-equal-util.ts

```diff
@@ -3,5 +3,9 @@ export function isTargetEqual<T extends TargetHost, U extends TargetHost>(
   a: T,
   b: U,
 ) {
-  return a.target === b.target;
+  return a.target === b.target || (
+    a.target.prototype ?
+      isTargetEqual({ target: (a.target as any).__proto__ }, b) :
+      false
+  );
```

Comment:
We might need to loop through prototypes here (to fully support the inheritance). The current solution looks like a workaround that won't work for longer inheritance chains.

Reply:
This works for longer inheritance chains because this function is applied to filter to get metadata properties of schemas.

Function reference:
https://github.com/nestjs/mongoose/blob/HEAD/lib/storages/type-metadata.storage.ts#L23-L29